# Marriott Hotel Category Change 2020 Analysis

In [1]:
# import pandas
import pandas as pd

In [2]:
# read data from csv
df = pd.read_csv('marriott-category-changes-2020.csv')
df.head()

Hotel                  Brand     Destination  \
0     Aberdeen Marriott Hotel               Marriott  United Kingdom   
1             AC Hotel Aitana  AC Hotels by Marriott           Spain   
2  AC Hotel Alcala de Henares  AC Hotels by Marriott           Spain   
3            AC Hotel Almeria  AC Hotels by Marriott           Spain   
4            AC Hotel Aravaca  AC Hotels by Marriott           Spain   

   Current Category  Current Standard Price  New Category  New Standard Price  
0                 3                   17500             2               12500  
1                 3                   17500             4               25000  
2                 1                    7500             2               12500  
3                 1                    7500             2               12500  
4                 1                    7500             2               12500

## Question 1
Discribe the data types for each feature/column, e.g., xxx feature's data type is String, yyy feature's data type is float, etc.

In [4]:
df.dtypes

Hotel                     object
Brand                     object
Destination               object
Current Category           int64
Current Standard Price     int64
New Category               int64
New Standard Price         int64
dtype: object

### ANSWER: The Hotel feature's data type is object, Brand feature's data type is object, Destination feature's data type is object, Current Category feature's data type is int64, Current Standard Price feature's data type is int64, New Category feature's data type is int64, and New Standard Price feature's data type is int64.

## Question 2
- How many hotels are in this dataset?
- The hotels are from how many unique brands?
- Which destination/country has the most hotels listed in this dataset? List the total number of hotels in that country
- How many brands in China have hotel category changes?

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2185 entries, 0 to 2184
Data columns (total 12 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Hotel                   2185 non-null   object
 1   Brand                   2185 non-null   object
 2   Destination             2185 non-null   object
 3   Current Category        2185 non-null   int64 
 4   Current Standard Price  2185 non-null   int64 
 5   New Category            2185 non-null   int64 
 6   New Standard Price      2185 non-null   int64 
 7   categorychange          2185 non-null   int64 
 8   change                  2185 non-null   object
 9   upgrade                 2185 non-null   object
 10  largedowngrade          2185 non-null   object
 11  largeupgrade            2185 non-null   object
dtypes: int64(5), object(7)
memory usage: 205.0+ KB


In [5]:
df.nunique()

Hotel                     2182
Brand                       30
Destination                 98
Current Category             8
Current Standard Price       8
New Category                 8
New Standard Price           8
dtype: int64

In [6]:
df.describe(include='all')

Hotel      Brand Destination  \
count                                2185       2185        2185   
unique                               2182         30          98   
top     Fairfield Inn & Suites Burlington  Courtyard         USA   
freq                                    2        366        1548   
mean                                  NaN        NaN         NaN   
std                                   NaN        NaN         NaN   
min                                   NaN        NaN         NaN   
25%                                   NaN        NaN         NaN   
50%                                   NaN        NaN         NaN   
75%                                   NaN        NaN         NaN   
max                                   NaN        NaN         NaN   

        Current Category  Current Standard Price  New Category  \
count        2185.000000             2185.000000   2185.000000   
unique               NaN                     NaN           NaN   
top                  NaN                     NaN           NaN   
freq                 NaN                     NaN           NaN   
mean            3.254462            21647.597254      3.797254   
std             1.441236            12020.922882      1.486691   
min             1.000000             7500.000000      1.000000   
25%             2.000000            12500.000000      3.000000   
50%             3.000000            17500.000000      3.000000   
75%             4.000000            25000.000000      5.000000   
max             8.000000            85000.000000      8.000000   

        New Standard Price  
count          2185.000000  
unique                 NaN  
top                    NaN  
freq                   NaN  
mean          26196.796339  
std           14964.210625  
min            7500.000000  
25%           17500.000000  
50%           17500.000000  
75%           35000.000000  
max           85000.000000

In [7]:
df.groupby('Destination')['Hotel'].count().max()

1548

In [8]:
import numpy as np
df['categorychange'] = df['Current Category'] - df['New Category']
df['change']= np.where(['categorychange'] != 0, '1', '0')
# change is 1 for yes, there was a category change, and 0 for no, there was not a category change.

In [9]:
df2 = df[(df['Destination'] == 'China') & (df['categorychange'] != 0)]
df2.groupby('Brand').count()
# total of 14 Brands below

Hotel  Destination  Current Category  \
Brand                                                        
Aloft                     2            2                 2   
Autograph Collection      1            1                 1   
Courtyard                 4            4                 4   
Delta                     1            1                 1   
Four Points               4            4                 4   
JW Marriott               5            5                 5   
Le Meridien               2            2                 2   
Luxury Collection         1            1                 1   
Marriott                 13           13                13   
Renaissance               4            4                 4   
Sheraton                 18           18                18   
St. Regis                 2            2                 2   
W Hotels                  3            3                 3   
Westin                    8            8                 8   

                      Current Standard Price  New Category  \
Brand                                                        
Aloft                                      2             2   
Autograph Collection                       1             1   
Courtyard                                  4             4   
Delta                                      1             1   
Four Points                                4             4   
JW Marriott                                5             5   
Le Meridien                                2             2   
Luxury Collection                          1             1   
Marriott                                  13            13   
Renaissance                                4             4   
Sheraton                                  18            18   
St. Regis                                  2             2   
W Hotels                                   3             3   
Westin                                     8             8   

                      New Standard Price  categorychange  change  
Brand                                                             
Aloft                                  2               2       2  
Autograph Collection                   1               1       1  
Courtyard                              4               4       4  
Delta                                  1               1       1  
Four Points                            4               4       4  
JW Marriott                            5               5       5  
Le Meridien                            2               2       2  
Luxury Collection                      1               1       1  
Marriott                              13              13      13  
Renaissance                            4               4       4  
Sheraton                              18              18      18  
St. Regis                              2               2       2  
W Hotels                               3               3       3  
Westin                                 8               8       8

### ANSWER: There are 2185 hotels in this dataset. The hotels are from a total of 30 different brands. USA has the most hotels listed in the dataset, with a total of 1548 hotels listed. There are 14 brands that have category changes in China.

## Question 3
- What's the percentage of hotels worldwide with category upgrade in 2020?

In [10]:
df['upgrade'] = df['New Category'] > df['Current Category']
df['upgrade']= np.where(df['New Category'] > df['Current Category'], 'True', 'False')
df[(df['upgrade'] == 'True')].count()

Hotel                     1686
Brand                     1686
Destination               1686
Current Category          1686
Current Standard Price    1686
New Category              1686
New Standard Price        1686
categorychange            1686
change                    1686
upgrade                   1686
dtype: int64

In [11]:
print(1686/2185)
# 1686 Hotel upgrades over 2185 total hotels

0.77162471395881


### 77.16% of hotels worldwide will have a category upgrade in 2020.

## Question 4
- List hotels with category changes greater than 1 if any, such as changing from category 3 to 5 or from category 7 to 4
- List all JW Marriott hotels in China that have a category upgrade

In [12]:
df['largedowngrade'] = np.where(df['categorychange'] > 1, 'true', 'false')
df['largeupgrade'] = np.where(df['categorychange'] < -1, 't', 'f')
df3 = df[(df['largeupgrade'] != 'f') | (df['largedowngrade'] != 'false')]
df3

Hotel        Brand Destination  \
913  Four Points by Sheraton Bali, Ungasan  Four Points   Indonesia   

     Current Category  Current Standard Price  New Category  \
913                 4                   25000             2   

     New Standard Price  categorychange change upgrade largedowngrade  \
913               12500               2      1   False           true   

    largeupgrade  
913            f

In [13]:
df6 = df[(df['Brand']=='JW Marriott') & (df['upgrade'] == 'True') & (df['Destination'] == 'China')]
df6

Hotel        Brand Destination  \
1074          JW Marriott Hotel Chengdu  JW Marriott       China   
1078         JW Marriott Hotel Shenzhen  JW Marriott       China   
1079  JW Marriott Hotel Shenzhen Bao'an  JW Marriott       China   
1083        JW Marriott Hotel Zhengzhou  JW Marriott       China   

      Current Category  Current Standard Price  New Category  \
1074                 3                   17500             4   
1078                 3                   17500             4   
1079                 3                   17500             4   
1083                 2                   12500             3   

      New Standard Price  categorychange change upgrade largedowngrade  \
1074               25000              -1      1    True          false   
1078               25000              -1      1    True          false   
1079               25000              -1      1    True          false   
1083               17500              -1      1    True          false   

     largeupgrade  
1074            f  
1078            f  
1079            f  
1083            f

### ANSWER: The Four Points by Sheraton Bali, Ungasan in Indonesia is the only hotel in the world that will change greater than one category. The hotel will change from category 4 to category 2. The JW Marriott hotels in China with category upgrades are Hotel Chengdu, Hotel Shenzhen, Hotel Shenzhen Bao'an, and Hotel Zhengzhou.

## Question 5
Assume you are in Feburary 2020 and the category changes will take effect on March 4, 2020. You are planning your trip to Florence, Italy and Hong Kong, China in April. You only stay in category 8 hotel (existing category 8 or future category 8) and want to optimize your point spending. Based on the data, which hotel you should book? when should you book your hotels for Florence and Hong Kong? Why?

In [14]:
df4 = df[(df['Current Category'] == 8) | (df['New Category'] == 8)].copy()
df5 = df4[(df4['Destination'] == 'Italy') | (df4['Destination'] == 'China')]
df5

Hotel              Brand  \
145                  Cervo Hotel, Costa Smeralda Resort           Sheraton   
519   Cristallo, a Luxury Collection Resort & Spa, C...  Luxury Collection   
1968                     The Westin Excelsior, Florence             Westin   
2165                                        W Hong Kong           W Hotels   

     Destination  Current Category  Current Standard Price  New Category  \
145        Italy                 7                   60000             8   
519        Italy                 8                   85000             7   
1968       Italy                 7                   60000             8   
2165       China                 8                   85000             7   

      New Standard Price  categorychange change upgrade largedowngrade  \
145                85000              -1      1    True          false   
519                60000               1      1   False          false   
1968               85000              -1      1    True          false   
2165               60000               1      1   False          false   

     largeupgrade  
145             f  
519             f  
1968            f  
2165            f

### ANSWER: Based on the results, you should book the W Hong Kong hotel first because it will be moving down from an 8 category to a 7. You should then book the Cristallo in Italy because it is currently a category 8 and will soon move down to a category 7. This will ensure that both of the hotels are category 8 when you are booked.